code/datums/manufacturing_requirements.dm

```diff
@@ -1,216 +1,338 @@
+			if (MATCH_ALL)
+				return (M.getMaterialFlags(M) & src.material_flags) == src.material_flags
+			if (MATCH_EXACT)
+				return (M.getMaterialFlags(M) == src.material_flags)
+
+/datum/manufacturing_requirement/match_flags/metal
+	name = "Metallic"
+	id = "metal_flag"
+	material_flags = MATERIAL_METAL
 
-/datum/manufacturing_requirement/match_property/wood
+/datum/manufacturing_requirement/match_flags/wood
 	name = "Wood"
-	id = "wood"
+	id = "wood_flag"
 	material_flags = MATERIAL_WOOD
+
+/datum/manufacturing_requirement/match_flags/rubber
+	name = "Rubber"
+	id = "rubber_flag"
+	material_flags = MATERIAL_RUBBER
+
+/datum/manufacturing_requirement/match_flags/organic_or_rubber
+	name = "Organic or Rubber"
+	id = "organic_or_rubber_flag"
+	material_flags = MATERIAL_ORGANIC | MATERIAL_RUBBER
+
+/datum/manufacturing_requirement/match_flags/fabric
+	name = "Fabric"
+	id = "fabric_flag"
+	material_flags = MATERIAL_RUBBER | MATERIAL_ORGANIC | MATERIAL_CLOTH
+
+/datum/manufacturing_requirement/match_flags/crystal
+	name = "Crystal"
+	id = "crystal_flag"
+	material_flags = MATERIAL_CRYSTAL
+
+/datum/manufacturing_requirement/match_flags/energy
+	name = "Energy Source"
+	id = "energy_flag"
+	material_flags = MATERIAL_ENERGY
+
+/datum/manufacturing_requirement/match_flags/insulated
+	name = "Insulative Material"
+	id = "insulative_flags"
+	material_flags = MATERIAL_CLOTH | MATERIAL_RUBBER
+
+#undef MATCH_ANY
+#undef MATCH_ALL
+#undef MATCH_EXACT
+
+ABSTRACT_TYPE(/datum/manufacturing_requirement/match_subtypes)
+/datum/manufacturing_requirement/match_subtypes
+	VAR_PROTECTED/match_typepath //! The parent type to use for istype() checks
+
+	#ifdef CHECK_MORE_RUNTIMES
+	New()
+		. = ..()
+		if (isnull(src.match_typepath) || !ispath(src.match_typepath))
+			CRASH("[src] has invalid match_typepath [src.match_typepath], it must be non-null and a valid path (not an instance!)")
+	#endif
+
+	is_match(var/datum/material/M)
+		. = ..()
+		if (!.) return
+		if (!(istype(M, src.match_typepath))) return FALSE
+
+/datum/manufacturing_requirement/match_subtypes/gemstone
+	name = "gemstone_subtypes"
+	id = "gemstone_subtypes"
+	match_typepath = /datum/material/crystal/gemstone
+
+/// Manufacturing requirements which check several conditions at once.
+ABSTRACT_TYPE(/datum/manufacturing_requirement/mixed)
+/datum/manufacturing_requirement/mixed
+	VAR_PROTECTED/list/datum/manufacturing_requirement/requirements = list() //! A list of requirements which must all be satisfied for this to return TRUE
+
+	/// Resolve the requirement paths to instances in the cache.
+	New()
+		. = ..()
+		var/list/datum/manufacturing_requirement/requirement_instances = list()
+		var/datum/manufacturing_requirement/requirement_id
+
+		for (var/datum/manufacturing_requirement/requirement_path as anything in src.requirements)
+			requirement_id = initial(requirement_path.id)
+			requirement_instances += getRequirement(requirement_id)
+		src.requirements = requirement_instances
```

Comment:
Why are we jumping through additional hoops by listing the requirements here as typepaths instead of by ID like everywhere else does?

Reply:
fair, i can bring it back to that